Sources/eppo/EppoClient.swift

```diff
@@ -102,25 +102,32 @@ public class EppoClient {
         host: String? = nil,
         assignmentLogger: AssignmentLogger? = nil,
         assignmentCache: AssignmentCache? = InMemoryAssignmentCache(),
-        initialConfiguration: Configuration? = nil
+        initialConfiguration: Configuration? = nil,
+        pollingEnabled: Bool = false,
+        pollingIntervalMs: Int = PollerConstants.DEFAULT_POLL_INTERVAL_MS,
+        pollingJitterMs: Int = PollerConstants.DEFAULT_POLL_INTERVAL_MS / PollerConstants.DEFAULT_JITTER_INTERVAL_RATIO
     ) async throws -> EppoClient {
         let instance = Self.initializeOffline(
-          sdkKey: sdkKey,
-          host: host,
-          assignmentLogger: assignmentLogger,
-          assignmentCache: assignmentCache,
-          initialConfiguration: initialConfiguration
+            sdkKey: sdkKey,
+            host: host,
+            assignmentLogger: assignmentLogger,
+            assignmentCache: assignmentCache,
+            initialConfiguration: initialConfiguration
         )
 
         return try await withCheckedThrowingContinuation { continuation in
-            initializerQueue.async {
```

Comment:
This is used to maintain the singleton pattern, although it's possible there's a more modern/elegant implementation: the SDK can be initialized in many places in the customer application. Please consult an AI assistant to see if the diff materially changes this.

Is this change needed to support polling?

Reply:
A test would be nice to demonstrate desired behavior, something like this:

```
func testSingletonInitializationConcurrency() async throws {
        // Call `initialize()` twice concurrently
        async let client1 = EppoClientManager.shared.initialize(
            sdkKey: "test-key"
        )
        async let client2 = EppoClientManager.shared.initialize(
            sdkKey: "test-key"
        )

        // Wait for both to complete
        let (result1, result2) = try await (client1, client2)

        // Assert that both results are the same instance
        XCTAssertTrue(result1 === result2, "Expected both calls to return the same singleton instance")
    }
```

Reply:
Not required for polling, but since polling needs to work with the rest of the SDK's concurrency, I paid a little extra attention to this compiler warning

![Screenshot 2025-04-22 at 10 57 07 PM](https://github.com/user-attachments/assets/e84a2f4d-b741-4cc6-8468-7c05a594df01)

Making `EppoClient` into a sendable type didn't seem like the right move as that could impact customer code, so then I questioned if this closure was necessary. In any case, a test sounds like a good idea. I can bring back the `initializerQueue`. It's probably harmless, though I'd prefer to not ignore warnings in case they indicate a risk

Reply:
Actually we already have tests for this

https://github.com/Eppo-exp/eppo-ios-sdk/blob/efdd1b19f66878d82a9558bc19883924800503ef/Tests/eppo/SingletonTest.swift

And they pass with and without the queue